src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CamundaService.java

```diff
@@ -89,8 +89,48 @@ public void claimTask(String taskId,
 
     }
 
-    public void assignTask(String taskId, String userId) {
+    public void assignTask(String taskId,
+                           AccessControlResponse assignerAccessControlResponse,
+                           List<PermissionTypes> assignerPermissionsRequired,
+                           AccessControlResponse assigneeAccessControlResponse,
+                           List<PermissionTypes> assigneePermissionsRequired) {
+        requireNonNull(
+            assigneeAccessControlResponse.getUserInfo().getUid(),
+            "AssigneeId cannot be null"
+        );
+
+        Map<String, CamundaVariable> variables = performGetVariablesAction(taskId);
+
+        boolean assignerHasAccess = permissionEvaluatorService.hasAccess(
+            variables,
+            assignerAccessControlResponse.getRoleAssignments(),
+            assignerPermissionsRequired
+        );
+
+        if (assignerHasAccess) {
+            boolean assigneeHasAccess = permissionEvaluatorService.hasAccess(
+                variables,
+                assigneeAccessControlResponse.getRoleAssignments(),
+                assigneePermissionsRequired
+            );
+            if (assigneeHasAccess) {
+                performAssignTaskAction(taskId, assigneeAccessControlResponse.getUserInfo().getUid());
+            } else {
+                throw new InsufficientPermissionsException(
+                    String.format("User did not have sufficient permissions to claim task with id: %s", taskId)
+                );
+            }
```

Comment:
This a very nitpick suggestion but you could get rid of the else but using a safe-guard instead.
It's fine as it is so up to you :) 

```suggestion
        if (!assigneeHasAccess) {
            throw new InsufficientPermissionsException(
                String.format("User did not have sufficient permissions to claim task with id: %s", taskId)
            );
        }
        performAssignTaskAction(taskId, assigneeAccessControlResponse.getUserInfo().getUid());
```

Reply:
Absolutely, good point. Done.